javaparser-core/src/main/java/com/github/javaparser/ast/Node.java

```diff
@@ -677,4 +683,91 @@ protected SymbolResolver getSymbolResolver() {
     // We need to expose it because we will need to use it to inject the SymbolSolver
     public static final DataKey<SymbolResolver> SYMBOL_RESOLVER_KEY = new DataKey<SymbolResolver>() {
     };
+
+
+    public enum TreeTraversal {
+        PREORDER, BREADTHFIRST, POSTORDER
+    }
+
+    public Iterator<Node> treeIterator(TreeTraversal traversal) {
```

Comment:
@ftomassetti a method to pick an iterator based on traversal algorithm

Reply:
nice!

Reply:
Perhaps you could move the logic into the TreeTraversal enum. Like:

```
enum TreeTraversal {
 BREADTHFIRST {
   Iterator<Node> iterator(Node node) { return TreeVisitor.BreadthFirstIterator(node); }
}
;
abstract Iterator<Node> iterator(Node node);
}
```

```
public Iterator<Node> treeIterator(TreeTraversal traversal) {
return traversal.iterator(this);
}
```

This would avoid the switch but maybe it is just overeegineering